util/keydecoder/keydecoder.go

```diff
@@ -0,0 +1,105 @@
+package keydecoder
```

Comment:
Remember to add the copyright headers in the new files.

Reply:
Todo: Check copyright headers in lint.